go.mod

```diff
@@ -22,6 +22,7 @@ require (
 	github.com/google/s2a-go v0.1.7 // indirect
 	github.com/googleapis/enterprise-certificate-proxy v0.3.2 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/stretchr/objx v0.5.0 // indirect
```

Comment:
please implement the feature without any external dependencies. you don't need `mock.Mock` - you can just implement a dummy type that has a `RoundTrip(request *http.Request) (*http.Response, error)` just for unit tests.

Reply:
Ok I can change this, although please note that the mock library (`github.com/stretchr/testify v1.8.4`) is already a part of the project, but for some reason its indirect dependency was not yet included. Let me know how would you like to proceed.

Reply:
simplest mock is 

```
type hc func(r *http.Request) (*http.Response, error)

func (cb hc) RoundTrip(r *http.Request) (*http.Response, error) {
	return cb(r)
}
```